packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -1299,7 +1326,10 @@ subtask(TASK_COMPILE_SOLIDITY_LOG_COMPILATION_RESULT)
 
       if (count > 0) {
         console.log(
-          `Compiled ${count} Solidity ${pluralize(count, "file")} successfully`
+          `Compiled ${count} Solidity ${pluralize(
+            count,
+            "file"
+          )} successfully with evmVersions: ${evmVersions.join(", ")}.`
         );
       }
     }
```

Comment:
I think you can accomplish the same thing you did but modifying just this subtask:

```js
subtask(TASK_COMPILE_SOLIDITY_LOG_COMPILATION_RESULT)
  .addParam("compilationJobs", undefined, undefined, types.any)
  .addParam("quiet", undefined, undefined, types.boolean)
  .setAction(
    async ({ compilationJobs }: { compilationJobs: CompilationJob[] }) => {
      let count = 0;
      const evmVersionsSet = new Set<string>();

      for (const job of compilationJobs) {
        const version = job.getSolcConfig().version;
        const evmVersion =
          job.getSolcConfig().settings.evmVersion ?? `default for ${version}`;
        evmVersionsSet.add(evmVersion);

        count += job
          .getResolvedFiles()
          .filter((file) => job.emitsArtifacts(file)).length;
      }

      const evmVersions = Array.from(evmVersionsSet);

      if (count > 0) {
        console.log(
          `Compiled ${count} Solidity ${pluralize(
            count,
            "file"
          )} successfully with evmVersions: ${evmVersions.join(", ")}.`
        );
      }
    }
  );
```

Or am I missing something here?

Reply:
Yes, not sure why I missed that. I moved the code.
But I kept the 2 sets to print the versions in order. I think that it looks nicer from a user prospective. What do you think?